vimiv/imutils/metadata.py

```diff
@@ -0,0 +1,304 @@
+                f"{', '.join(failed)}<br>\n"
+                f"Some metadata may be missing in the destination image {dest}."
+            )
+
+    def get_date_time(self) -> str:
+        """Get creation date and time as formatted string.
+
+        Raises:
+            UnsupportedMetadataOperation if no implementation supports this operation.
+        """
+
+        if not has_get_date_time():
+            MetadataHandler.raise_exception()
+
+        for _, backend in _registry.items():
+            with contextlib.suppress(NotImplementedError):
+                return backend(self._path).get_date_time()
+        return ""
+
+    def get_metadata(self, keys: Sequence[str]) -> MetadataDictT:
+        """Get value of all desired keys.
+
+        Use all registered implementations for `Operations.get_metadata` to query
+        the current image using the provided metadata keys. The result of all
+        implementations is combined.
+
+        Args:
+            keys: Keys of metadata to query the image for.
+
+        Returns:
+            Dictionary with retrieved metadata.
+
+        Raises:
+            UnsupportedMetadataOperation if no implementation supports this operation.
+        """
+
+        if not has_get_metadata():
+            MetadataHandler.raise_exception()
+
+        out: MetadataDictT = {}
+
+        for _, backend in _registry.items():
+            with contextlib.suppress(NotImplementedError):
+                # TODO: from 3.9 on use: c = a | b
+                out = {**backend(self._path).get_metadata(keys), **out}
+
+        return out
+
+    def get_keys(self) -> Iterable[str]:
+        """Retrieve the key of all metadata values available in the current image.
+
+        Raises:
+            UnsupportedMetadataOperation if no implementation supports this operation.
+        """
+
+        if not has_get_keys():
+            MetadataHandler.raise_exception()
+
+        out: Iterable[str] = iter([])
+
+        for _, backend in _registry.items():
+            with contextlib.suppress(NotImplementedError):
+                out = itertools.chain(backend(self._path).get_keys(), out)
+
+        return out
+
+    @staticmethod
+    def raise_exception() -> NoReturn:
+        """Raise an exception for a operations without implementation."""
+        msg = f"{inspect.stack()[1][3]} has no implementation"
+        _logger.warning(msg, once=True)
+        raise UnsupportedMetadataOperation(msg)
+
+
+class UnsupportedMetadataOperation(NotImplementedError):
+    """Raised if for an Operations, no function implementation is registered."""
+
+
+def register(plugin: Type[MetadataPlugin]) -> None:
+    """Register metadata functionality for usage by MetadataHandler.
+
+    Args:
+        plugin: Subclass of MetadataPlugin that implements metadat functionality.
+    """
+    assert plugin.name, "Required to have `plugin.name` set"
+    assert plugin.version, "Required to have `plugin.version` set"
```

Comment:
They are not needed. They are just for the static analysis. I can also just ignore the warning if you prefere:
```
_registry[plugin.name] = plugin  # type: ignore[index]
```

Reply:
Ah, because `mypy` doesn't know `MetadataPlugin` must actually defines these I guess? Would probably `type: ignore` then with a quick comment explaining this :blush:

Reply:
Yes exactly.

Is there any particular reason why you do not specify the exact code/error to ignore? I.e. why `type: ignore` over `type: ignore[index]`?

Reply:
Bad habits if anything in the code, laziness when typing here :laughing:

Reply:
Haha, makes sense :upside_down_face: 

This is no longer relevant due to the refactoring in aafe415fb7230ed7b63342eee41de7eeafdb190d